src/lib/ibcclient.ts

```diff
@@ -246,19 +249,21 @@ export class IbcClient {
       setupIbcExtension
     );
     this.senderAddress = senderAddress;
-    const { gasPrice = defaultGasPrice, gasLimits = {} } = options;
+    const { gasPrice = defaultGasPrice, gasLimits = {}, logger } = options;
     this.fees = buildFeeTable<IbcFeeTable>(
       gasPrice,
       defaultGasLimits,
       gasLimits
     );
+    this.logger = logger;
   }
 
   public getChainId(): Promise<string> {
     return this.sign.getChainId();
   }
 
   public async latestHeader(): Promise<RpcHeader> {
+    this.logger?.info('Getting latest header');
```

Comment:
I guess we can use `?` everywhere instead of NoopLogger. I guess not much writing

Reply:
I have no preference here, it's just a matter of style.